### DISCUSSION TITLE
Prevent All Re-Ordering for Expressions (besides when calling simplify)

### DISCUSSION THREAD

**Discussion by swiftly-viper** [sympy/sympy#26909] [General]
Hello,

Does anyone know how we can prevent unwanted term reordering when manipulating expressions (Expr class), which comprises various subclasses such as Derivative when using the .subs method?

For example:

If I have the Expr:

Expr(Derivative(exp(x**2), x))

and I call the .subs method on it, applying the Chain Rule

Expr(Derivative(exp(x**2, x)).subs(Derivative(exp(x**2, x)), Derivative(exp(u), u) * Derivative(x**2, x))

Currently, a print of the above would produce:

Derivative(x**2, x)) * Derivative(exp(u), u)

I want to prevent all re-ordering of terms. Is the issue using .subs since it calls simplify? I am not sure. If there is an alternative let me know.

**Comment by oscarbenjamin** [comment]
GitHub issues are not for getting help with using SymPy. I'm going to turn this into a discussion.

**Reply by swiftly-viper** [reply]
Apologies for the mistake, and thank you for the correction.
